crates/node-file-trace/src/lib.rs

```diff
@@ -493,8 +452,15 @@ async fn run<B: Backend + 'static, F: Future<Output = ()>>(
         let dir = dir.clone();
         let args = args.clone();
         let sender = sender.clone();
+        let module_options = module_options.clone();
+        let resolve_options = resolve_options.clone();
         Box::pin(async move {
-            let output = main_operation(TransientValue::new(dir.clone()), args.clone().into());
+            let output = main_operation(
+                TransientValue::new(dir.clone()),
+                args.clone().into(),
+                module_options.map(|m| m.into()),
+                resolve_options.map(|r| r.into()),
```

Comment:
You `cell` these values into the root task. And there will be a new root task for every call to `run`. So everything see different instances of `module_options` and `resolve_options` and they can't be shared.

You also don't want to `cell` stuff in `main_operation` for the same reason.

You want to pull all these `AssetContextVc` creation related stuff into a separate method and get passed `module_options` and `resolve_options` by value. (`Value<ModuleOptionsContext>`)

Reply:
`Value<ModuleOptionsContext>` is unable to pass `#[turbo_tasks::function]` because it lack of implementation for `Hash, Ord` and it's hard to implement these trait for `ModuleOptionsContext` and `ResolveOptionsContext` because they contain many other structs in their type definition.

Reply:
Let's pass in `ModuleOptionsContextVc` and `ResolveOptionsContextVc` in the first place as the arguments of `fn run` if we want to share them.